### PR TITLE
[Docs] Further explained the mythical `InvalidPacket`

### PR DESCRIPTION
**Pull request by Jarno458**

### CODE REVIEW

File: docs/network protocol.md [hunk 1]
```diff
@@ -191,6 +191,21 @@ Sent to clients after a client requested this message be sent to them, more info
...
 
+#### Arguments
+| Name | Type | Notes |
+| ---- | ---- | ----- |
+| type | str | The [PacketProblemType](#PacketProblemType) that was detected in the packet |
+| original_cmd | Optional[str] | the `cmd` argument of the faulty packet, will be `None` if the `cmd` failed to be parsed |
+| text | str | A descriptive message of the problem at hand |
+
+##### PacketProblemType
+PacketProblemType indicates the type problem that was detected in the faulty packet, the known problem types are below but others maybe added in the future
```
Comment (black-sliver) [inline docs/network protocol.md:202]:
```suggestion
PacketProblemType indicates the type problem that was detected in the faulty packet, the known problem types are below but others may be added in the future.
```

File: docs/network protocol.md [hunk 2]
```diff
@@ -191,6 +191,21 @@ Sent to clients after a client requested this message be sent to them, more info
 ### InvalidPacket
 Sent to clients if the server caught a problem with a packet. This only occurs for errors that are explicitly checked for.
 
+#### Arguments
+| Name | Type | Notes |
+| ---- | ---- | ----- |
+| type | str | The [PacketProblemType](#PacketProblemType) that was detected in the packet |
+| original_cmd | Optional[str] | the `cmd` argument of the faulty packet, will be `None` if the `cmd` failed to be parsed |
+| text | str | A descriptive message of the problem at hand |
```
Comment (black-sliver) [inline docs/network protocol.md:199]: descriptive like `'text': 'Connect'` ? :D
Maybe we should rework the messages one day to make it mach the protol.md :)

**Review by black-sliver — CHANGES_REQUESTED** [review]

File: docs/network protocol.md (hunk 2 above)
Comment (Jarno458) [reply docs/network protocol.md:199]: Yeah some are more descriptive than others, but I write it down as the way i think it was meant to be

File: docs/network protocol.md [hunk 3]
```diff
@@ -191,6 +191,21 @@ Sent to clients after a client requested this message be sent to them, more info
...
+
+##### PacketProblemType
+PacketProblemType indicates the type problem that was detected in the faulty packet, the known problem types are below but others may be added in the future.
+
+| Type | Notes |
+| ---- | ----- |
+| cmd | signaling that the `cmd` argument of the faulty packet could not be parsed correctly |
+| arguments | signaling that the arguments of the faulty packet are not correct |
```
Comment (black-sliver) [inline docs/network protocol.md:207]: If there is a reason to not make this change, let me know and i will approve as is
```suggestion
| cmd | `cmd` argument of the faulty packet could not be parsed correctly |
| arguments | arguments of the faulty packet are not correct |
```

**Review by black-sliver — CHANGES_REQUESTED** [review]

File: docs/network protocol.md [hunk 4]
```diff
@@ -191,6 +191,21 @@ Sent to clients after a client requested this message be sent to them, more info
 ### InvalidPacket
 Sent to clients if the server caught a problem with a packet. This only occurs for errors that are explicitly checked for.
 
+#### Arguments
+| Name | Type | Notes |
+| ---- | ---- | ----- |
+| type | str | The [PacketProblemType](#PacketProblemType) that was detected in the packet |
```
Comment (Ijwu) [inline docs/network protocol.md:197]:
```suggestion
| type | str | The [PacketProblemType](#PacketProblemType) that was detected in the packet. |
```

File: docs/network protocol.md [hunk 5]
```diff
@@ -191,6 +191,21 @@ Sent to clients after a client requested this message be sent to them, more info
...
+| original_cmd | Optional[str] | the `cmd` argument of the faulty packet, will be `None` if the `cmd` failed to be parsed |
```
Comment (Ijwu) [inline docs/network protocol.md:198]:
```suggestion
| original_cmd | Optional[str] | The `cmd` argument of the faulty packet, will be `None` if the `cmd` failed to be parsed. |
```

File: docs/network protocol.md (hunk 2 above)
Comment (Ijwu) [inline docs/network protocol.md:199]:
```suggestion
| text | str | A descriptive message of the problem at hand. |
```

File: docs/network protocol.md [hunk 6]
```diff
@@ -191,6 +191,21 @@ Sent to clients after a client requested this message be sent to them, more info
...
+| text | str | A descriptive message of the problem at hand |
+
+##### PacketProblemType
+PacketProblemType indicates the type problem that was detected in the faulty packet, the known problem types are below but others may be added in the future.
+
+| Type | Notes |
+| ---- | ----- |
+| cmd | `cmd` argument of the faulty packet could not be parsed correctly |
```
Comment (Ijwu) [inline docs/network protocol.md:206]:
```suggestion
| cmd | `cmd` argument of the faulty packet that could not be parsed correctly. |
```

File: docs/network protocol.md [hunk 7]
```diff
@@ -191,6 +191,21 @@ Sent to clients after a client requested this message be sent to them, more info
...
+| arguments | arguments of the faulty packet are not correct |
```
Comment (Ijwu) [inline docs/network protocol.md:207]:
```suggestion
| arguments | Arguments of the faulty packet which were not correct. |
```

File: docs/network protocol.md [hunk 8]
```diff
@@ -191,6 +191,21 @@ Sent to clients after a client requested this message be sent to them, more info
...
 
+#### Arguments
+| Name | Type | Notes |
+| ---- | ---- | ----- |
+| type | str | The [PacketProblemType](#PacketProblemType) that was detected in the packet |
+| original_cmd | Optional[str] | the `cmd` argument of the faulty packet, will be `None` if the `cmd` failed to be parsed |
+| text | str | A descriptive message of the problem at hand |
+
+##### PacketProblemType
+PacketProblemType indicates the type problem that was detected in the faulty packet, the known problem types are below but others may be added in the future.
```
Comment (Ijwu) [inline docs/network protocol.md:202]:
```suggestion
`PacketProblemType` indicates the type of problem that was detected in the faulty packet, the known problem types are below but others may be added in the future.
```

File: docs/network protocol.md [hunk 9]
```diff
@@ -191,6 +191,21 @@ Sent to clients after a client requested this message be sent to them, more info
...
+
+| Type | Notes |
+| ---- | ----- |
+| cmd | `cmd` argument of the faulty packet could not be parsed correctly |
+| arguments | arguments of the faulty packet are not correct |
+
 ### Retrieved
 Sent to clients as a response the a [Get](#Get) package
```
Comment (Ijwu) [inline docs/network protocol.md:210]:
```suggestion
Sent to clients as a response the a [Get](#Get) package.
```

Yeah, I know this isn't your change. But I'm suggesting it anyway.

**Review by Ijwu — CHANGES_REQUESTED** [review]
tfw no punctuation

File: docs/network protocol.md (hunk 2 above)
Comment (black-sliver) [reply docs/network protocol.md:199]: ^ that's not a complete sentence (which is why i suggested removing the 3 words on the front)

**Review by Ijwu — APPROVED** [review]

**Review by black-sliver — APPROVED** [review]

**Pull request merged**